src/metakb/cli.py

```diff
@@ -14,178 +14,384 @@
 import boto3
 from boto3.exceptions import ResourceLoadException
 from botocore.config import Config
-from disease.cli import update_db as update_normalizer_disease_db
 from disease.database.database import AWS_ENV_VAR_NAME as DISEASE_AWS_ENV_VAR_NAME
-from gene.cli import update_normalizer_db as update_normalizer_gene_db
 from gene.database.database import AWS_ENV_VAR_NAME as GENE_AWS_ENV_VAR_NAME
-from therapy.cli import update_normalizer_db as update_normalizer_therapy_db
 from therapy.database.database import AWS_ENV_VAR_NAME as THERAPY_AWS_ENV_VAR_NAME
 
-from metakb import APP_ROOT
+from metakb import APP_ROOT, DATE_FMT
 from metakb.database import Graph
 from metakb.harvesters.civic import CivicHarvester
 from metakb.harvesters.moa import MoaHarvester
+from metakb.normalizers import NormalizerName, ViccNormalizers, update_normalizer
+from metakb.normalizers import check_normalizers as check_normalizer_health
 from metakb.schemas.app import SourceName
 from metakb.transform import CivicTransform, MoaTransform
 
-logger = logging.getLogger(__name__)
+logging.basicConfig(
+    filename=f"{__name__}.log",
+    format="%(asctime)s %(levelname)s:%(name)s:%(message)s",
+    force=True,
+)
+_logger = logging.getLogger(__name__)
 
 
-def echo_info(msg: str) -> None:
+def _echo_info(msg: str) -> None:
     """Log (as INFO) and echo given message.
 
     :param msg: message to emit
     """
     click.echo(msg)
-    logger.info(msg)
+    _logger.info(msg)
 
 
-@click.command()
-@click.option(
-    "--db_url",
-    default="bolt://localhost:7687",
-    help=(
-        "URL endpoint for the application Neo4j database. Can also be provided via environment variable METAKB_DB_URL, which takes priority."
-    ),
-)
-@click.option(
-    "--db_username",
-    default="neo4j",
-    help=(
-        "Username to provide to application Neo4j database. Can also be provided via environment variable METAKB_DB_USERNAME, which takes priority."
-    ),
+def _help_msg(msg: str = "") -> None:
+    """Handle invalid user input.
+
+    :param msg: Error message to display to user.
+    """
+    ctx = click.get_current_context()
+    _logger.fatal(msg)
+
+    if msg:
+        click.echo(msg)
+    else:
+        click.echo(ctx.get_help())
+
+    ctx.exit()
+
+
+@click.group()
+def cli() -> None:
+    """Manage MetaKB data.
+
+    To prepare normalizers if unavailable, invalidate cached data, then load MetaKB data:
+
+    \b
+        $ metakb check-normalizers || metakb load-normalizers
+        $ metakb update --update_source_caches
+    """  # noqa: D301
+
+
+_normalizer_db_url_description = "URL endpoint of normalizer database. If not given, the individual normalizers will revert to their own defaults."
```

Comment:
Since Dynamo takes a single URL but Postgres requires different libpq-style URLs that include normalizer-specific database names, I don't think it's really feasible to support postgres connections this way yet. Thought about doing something that could take a base postgres URL and add database names on top but it's probably not worth the work right now.